internal/authorizedapp/memory_provider.go

```diff
@@ -27,26 +29,44 @@ var _ Provider = (*MemoryProvider)(nil)
 // MemoryProvider is an Provider that stores values in-memory. It is primarily
 // used for testing.
 type MemoryProvider struct {
-	Data map[string]*model.AuthorizedApp
+	lock sync.RWMutex
+	data map[string]*model.AuthorizedApp
```

Comment:
this is just for tests right, why not have tests use the test DB infrastructure?

Reply:
This is faster and cheaper.